src/main.py

```diff
@@ -159,15 +164,79 @@ def main():
             os.path.expanduser(args.assistant_secrets))
         recognizer = speech.AssistantSpeechRequest(credentials)
 
-    recorder = audio.Recorder(
-        input_device=args.input_device, channels=1,
-        bytes_per_sample=speech.AUDIO_SAMPLE_SIZE,
-        sample_rate_hz=speech.AUDIO_SAMPLE_RATE_HZ)
-    with recorder:
-        do_recognition(args, recorder, recognizer, player)
+    status_ui = StatusUi(player, args.led_fifo, args.trigger_sound)
 
+    # The ok-google trigger is handled with the Assistant Library, so we need
+    # to catch this case early.
+    if args.trigger == 'ok-google':
```

Comment:
@proppy not sure if it helps; but, I pulled down this branch and tested it.  is this the `start-conversation` you are talking about?

> "Hey Google, how far away is Japan?"
< "Japan is 6xxx miles away as way of the crow flies."
> "Hey Google, and from California?"
< "Japan is 5xxx miles away from California."

```
$ src/main.py --trigger="ok-google"
[2017-05-26 22:25:17,165] INFO:root:ON_MUTED_CHANGED:
  {'is_muted': False}
[2017-05-26 22:25:17,166] INFO:root:ON_START_FINISHED
[2017-05-26 22:25:17,167] INFO:main:ready...
Say "OK, Google" then speak, or press Ctrl+C to quit...
[2017-05-26 22:25:24,515] INFO:root:ON_CONVERSATION_TURN_STARTED
[2017-05-26 22:25:24,516] INFO:main:listening...
[2017-05-26 22:25:28,365] INFO:root:ON_END_OF_UTTERANCE
[2017-05-26 22:25:28,366] INFO:main:thinking...
[2017-05-26 22:25:28,385] INFO:root:ON_RECOGNIZING_SPEECH_FINISHED:
  {'text': 'how far away is Japan'}
[2017-05-26 22:25:28,907] INFO:root:ON_RESPONDING_STARTED:
  {'is_error_response': False}
[2017-05-26 22:25:33,156] INFO:root:ON_RESPONDING_FINISHED
[2017-05-26 22:25:33,157] INFO:root:ON_CONVERSATION_TURN_FINISHED:
  {'with_follow_on_turn': False}
[2017-05-26 22:25:33,158] INFO:main:ready...
[2017-05-26 22:25:34,972] INFO:root:ON_CONVERSATION_TURN_STARTED
[2017-05-26 22:25:34,974] INFO:main:listening...
[2017-05-26 22:25:37,748] INFO:root:ON_END_OF_UTTERANCE
[2017-05-26 22:25:37,749] INFO:main:thinking...
[2017-05-26 22:25:37,784] INFO:root:ON_RECOGNIZING_SPEECH_FINISHED:
  {'text': 'and from California'}
[2017-05-26 22:25:38,856] INFO:root:ON_RESPONDING_STARTED:
  {'is_error_response': False}
[2017-05-26 22:25:42,704] INFO:root:ON_RESPONDING_FINISHED
[2017-05-26 22:25:42,705] INFO:root:ON_CONVERSATION_TURN_FINISHED:
  {'with_follow_on_turn': False}
[2017-05-26 22:25:42,707] INFO:main:ready...
```

Also, I played the Trivia game and answered five questions in turn.  

> "Hey Google, let's play a game."